draft-ietf-quic-transport.md

```diff
@@ -2196,46 +2230,40 @@ new stream.
-stream limit of 0 will cause a connection to be unusable.  Application protocols
-that use QUIC might require a certain minimum number of streams to function
-correctly.  If a peer advertises an concurrent stream limit (concurrent_streams)
-that is too small for the selected application protocol to function, an endpoint
-MUST terminate the connection with an error of type QUIC_TOO_MANY_OPEN_STREAMS
-({{error-handling}}).
+A QUIC endpoint cannot reuse a StreamID on a given connection.  Streams MUST be
+created in sequential order.  Open streams can be used in any order.
 
 
 ## Stream Concurrency
 
-An endpoint limits the number of concurrently active incoming streams by setting
-the concurrent stream limit (see {{transport-parameter-definitions}}) in the
-transport parameters. The maximum concurrent streams setting is specific to each
-endpoint and applies only to the peer that receives the setting. That is,
-clients specify the maximum number of concurrent streams the server can
-initiate, and servers specify the maximum number of concurrent streams the
-client can initiate.
-
-Streams that are in the "open" state or in either of the "half-closed" states
-count toward the maximum number of streams that an endpoint is permitted to
-open.  Streams in any of these three states count toward the limit advertised in
-the concurrent stream limit.
-
-A recently closed stream MUST also be considered to count toward this limit
-until packets containing all frames required to close the stream have been
-acknowledged. For a stream which closed cleanly, this means all STREAM frames
-have been acknowledged; for a stream which closed abruptly, this means the
-RST_STREAM frame has been acknowledged.
+An endpoint limits the number of concurrently active incoming streams by
+adjusting the maximum stream ID.  An initial value is set in the transport
+parameters  (see {{transport-parameter-definitions}}) and is subsequently
+increased by LIMIT_UPDATE frames (see {{frame-limit-update}}). The maximum
+stream ID is specific to each endpoint and applies only to the peer that
+receives the setting. That is, clients specify the maximum stream ID the server
+can initiate, and servers specify the maximum stream ID the client can initiate.
+Each endpoint may respond on streams initiated by the other peer, regardless of
+whether it is permitted to initiated new streams.
 
 Endpoints MUST NOT exceed the limit set by their peer.  An endpoint that
-receives a STREAM frame that causes its advertised concurrent stream limit to be
-exceeded MUST treat this as a stream error of type QUIC_TOO_MANY_OPEN_STREAMS
-({{error-handling}}).
-
+receives a STREAM frame with an ID greater than the limit it has sent MUST treat
+this as a stream error of type QUIC_TOO_MANY_OPEN_STREAMS ({{error-handling}}),
+unless this is a result of a change in the initial offsets (see
+{{zerortt-parameters}}).
+
+A Maximum Stream ID of 0 will cause a connection to be unusable.  Application
```

Comment:
1 will likely make any application protocol unusable, either because it requires a control stream to open immediately (HTTP) or because the connection will be unable to serve any requests.  However, for something like the "connect and PING" test you proposed for first-interop (maybe ALPN token of "none"?), "1" would be an appropriate value to advertise.